src/service/resources-impl.js

```diff
@@ -49,6 +50,16 @@ const MUTATE_DEFER_DELAY_ = 500;
 const FOCUS_HISTORY_TIMEOUT_ = 1000 * 60;  // 1min
 const FOUR_FRAME_DELAY_ = 70;
 
+/**
+ * The internal state of the visibility state machine.
+ * @typedef {{
+ *   viewer: boolean,
```

Comment:
In this case we are pretty strict. We can always switch state machine to require a string or a number too - I'm sure we can live without extra stringifies :) In all seriousness though - this is a critical path code, so if we can avoid extra actions, we should.